audit-cli/README.md

```diff
@@ -0,0 +1,876 @@
+# audit-cli
+
+A Go CLI tool for extracting and analyzing code examples from MongoDB documentation written in reStructuredText (RST).
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Installation](#installation)
+- [Usage](#usage)
+  - [Extract Commands](#extract-commands)
+  - [Search Commands](#search-commands)
+  - [Analyze Commands](#analyze-commands)
+- [Development](#development)
+  - [Project Structure](#project-structure)
+  - [Adding New Commands](#adding-new-commands)
+  - [Testing](#testing)
+  - [Code Patterns](#code-patterns)
+- [Supported RST Directives](#supported-rst-directives)
+
+## Overview
+
+This CLI tool helps maintain code quality across MongoDB's documentation by:
+
+1. **Extracting code examples** from RST files into individual, testable files
+2. **Searching extracted code** for specific patterns or substrings
+3. **Analyzing include relationships** to understand file dependencies
+4. **Following include directives** to process entire documentation trees
+5. **Handling MongoDB-specific conventions** like steps files, extracts, and template variables
+
+## Installation
+
+### Build from Source
+
+```bash
+cd audit-cli
+go build
+```
+
+This creates an `audit-cli` executable in the current directory.
+
+### Run Without Building
+
+```bash
+cd audit-cli
+go run main.go [command] [flags]
+```
+
+## Usage
+
+The CLI is organized into parent commands with subcommands:
+
+```
+audit-cli
+├── extract          # Extract content from RST files
+│   └── code-examples
+├── search           # Search through extracted content
+│   └── find-string
+└── analyze          # Analyze RST file structures
+    └── includes
+```
+
+### Extract Commands
+
+#### `extract code-examples`
+
+Extract code examples from reStructuredText files into individual files.
+
+**Basic Usage:**
+
+```bash
+# Extract from a single file
+./audit-cli extract code-examples path/to/file.rst -o ./output
+
+# Extract from a directory (non-recursive)
+./audit-cli extract code-examples path/to/docs -o ./output
+
+# Extract recursively from all subdirectories
+./audit-cli extract code-examples path/to/docs -o ./output -r
+
+# Follow include directives
+./audit-cli extract code-examples path/to/file.rst -o ./output -f
+
+# Combine recursive scanning and include following
+./audit-cli extract code-examples path/to/docs -o ./output -r -f
+
+# Dry run (show what would be extracted without writing files)
+./audit-cli extract code-examples path/to/file.rst -o ./output --dry-run
+
+# Verbose output
+./audit-cli extract code-examples path/to/file.rst -o ./output -v
+```
+
+**Flags:**
+
+- `-o, --output <dir>` - Output directory for extracted files (default: `./output`)
+- `-r, --recursive` - Recursively scan directories for RST files
+- `-f, --follow-includes` - Follow `.. include::` directives in RST files
```

Comment:
this might benefit from more detailed description or a specific use case (alternatively, when you _don't_ want to use this)

also, how does it interact with `-r`?